app/models/abuse_report.rb

```diff
@@ -39,42 +39,57 @@ def akismet_attributes
 
   scope :by_date, -> { order('created_at DESC') }
 
-  before_validation :add_work_id_to_url, :clean_url, on: :create
-  
-  # Clean work or profile URLs so we can prevent the same URLs from
-  # getting reported too many times.
-  # If the URL ends without a / at the end, add it:
-  # url_is_not_over_reported uses the / so "/works/1234" isn't a match
-  # for "/works/123"
-  def clean_url
-    # Work URLs: "works/123"
-    # Profile URLs: "users/username"
-    if url =~ /(works\/\d+)/ || url =~ /(users\/\w+)/
-      uri = Addressable::URI.parse url
-      uri.query = nil
-      uri.fragment = nil
-      uri.path += "/" unless uri.path.end_with? "/"
-      self.url = uri.to_s
-    else
-      url
-    end
+  # Standardize the format of work, chapter, and profile URLs to get it ready
+  # for the url_is_not_over_reported validation.
+  # Work URLs: "works/123"
+  # Chapter URLs: "chapters/123"
+  # Profile URLs: "users/username"
+  before_validation :standardize_url, on: :create
+  def standardize_url
+    return unless url =~ %r{((chapters|works)/\d+)} || url =~ %r{(users\/\w+)}
+
+    self.url = add_scheme_to_url(url)
+    self.url = clean_url(url)
+    self.url = add_work_id_to_url(self.url)
+  end
+
+  def add_scheme_to_url(url)
```

Comment:
You can combine this with `clean_url` so you have to run `Addressable::URI.parse` one less time. When you have the parsed uri object I think you can do `uri.scheme = "https"` if it doesn't have a scheme.

Reply:
Unfortunately, that didn't work. If you have a URL without a scheme and try to add one with `uri.scheme`, you just get, e.g., `https:archiveofourown.org`.  It has something to do with it not properly recognizing the host and treating it like a path, I think?

```ruby
irb(main):012:0> url = "archiveofourown.org/chapters/000"
=> "archiveofourown.org/chapters/000"
irb(main):013:0> uri = Addressable::URI.parse(url)
=> #<Addressable::URI:0x11044 URI:archiveofourown.org/chapters/000>
irb(main):014:0> uri.scheme = "https"
=> "https"
irb(main):015:0> uri.to_s
=> "https:archiveofourown.org/chapters/000"
irb(main):016:0> uri.host
=> nil
irb(main):017:0> uri.path
=> "archiveofourown.org/chapters/000"
```

Reply:
Gotcha, then this is fine as is.